src/index.ts

```diff
@@ -1,89 +1,148 @@
+mediatorConfig.config.fhirMappings = JSON.stringify(fhirMappings, null, 2)
 
-const plugins: Map<string, FhirPlugin> = GetFhirPlugins(fhirMappings);
+const plugins: Map<string, FhirPlugin> = GetFhirPlugins(fhirMappings)
 
-// TODO: find out if http is required for local testing
-const kafkaHost = process.env.KAFKA_HOST || "http://localhost";
-const kafkaPort = process.env.KAFKA_PORT || "9092";
+const kafkaHost = process.env.KAFKA_HOST ?? 'localhost'
+const kafkaPort = process.env.KAFKA_PORT ?? '9092'
+const trustSelfSigned = (process.env.TRUST_SELF_SIGNED ?? 'false') === 'true'
+const openhimAuth = {
+  apiURL: process.env.OPENHIM_API_URL ?? 'https://localhost:8080',
+  username: process.env.OPENHIM_USERNAME ?? 'root@openhim.org',
+  password: process.env.OPENHIM_PASSWORD ?? 'password',
+  trustSelfSigned,
+  urn: 'urn:openhim-mediator:kafka-mapper-consumer'
+}
+const shouldRegisterMediator =
+  (process.env.REGISTER_MEDIATOR ?? 'false') === 'true'
+const fromBeginning = (process.env.KAFKA_FROM_BEGINNING ?? 'true') === 'true'
+const consumerGroupId = process.env.CONSUMER_GROUP_ID ?? 'kafka-mapper-consumer'
 
 const kafka = new Kafka({
   logLevel: logLevel.INFO,
   brokers: [`${kafkaHost}:${kafkaPort}`],
-  clientId: "kafka-mapper-consumer",
-});
+  clientId: 'kafka-mapper-consumer'
+})
+
+let consumer = kafka.consumer({groupId: 'kafka-mapper-consumer'})
+
+const run = async (newFhirMappings?) => {
+  if (newFhirMappings) {
+    await consumer.disconnect()
+    // create new consumer so we can subscribe to a (potentially) different set of topics
+    consumer = kafka.consumer({groupId: consumerGroupId})
+    fhirMappings = newFhirMappings
+  }
 
-const consumer = kafka.consumer({ groupId: "kafka-mapper-consumer" });
-const producer = kafka.producer();
+  // Register as an OpenHIM mediator and use that for config management
+  if (shouldRegisterMediator && !newFhirMappings) {
+    console.log(
+      'Registering as an OpenHIM Mediator. The OpenHIM will be used to supply latest config not the config file, the config file will provide the default config to the OpenHIM.'
+    )
+    await promisify(registerMediator)(openhimAuth, mediatorConfig)
+    const configEmitter = activateHeartbeat(openhimAuth)
 
-const run = async () => {
-  const topics = fhirMappings.map((fhirMapping) => fhirMapping.resourceType.toLowerCase());
+    configEmitter.on('config', config => {
+      console.log('Received new mapping config, restarting consumer...')
+      // restart consumer with new config
+      run(JSON.parse(config.fhirMappings))
+    })
 
-  await consumer.connect();
-  await producer.connect();
-  await consumer.subscribe({ topics, fromBeginning: true });
+    const config = await promisify(fetchConfig)(openhimAuth)
+    // set config to that received from the OpenHIM
+    fhirMappings = JSON.parse(config.fhirMappings)
+  }
+
+  const fhirMappingValidationErrors = ValidateFhirMappingsJson(fhirMappings)
```

Comment:
```suggestion
  const fhirMappingValidationErrors = validateFhirMappingsJson(fhirMappings)
```

Reply:
At this line we are not sure if fhirMappings is not empty (shouldRegisterMediator = false && newFhirMappings = false)

Reply:
It won't be empty as it is always set to the default of what is in the default config file. See line 21.